traefik/templates/service.yaml

```diff
@@ -3,9 +3,10 @@
 {{ $tcpPorts := dict }}
 {{ $udpPorts := dict }}
 {{- range $name, $config := .Values.ports }}
-  {{- if eq (toString $config.protocol) "UDP" }}
+  {{- if or $config.http3 (eq (toString $config.protocol) "UDP") }}
     {{ $_ := set $udpPorts $name $config }}
-  {{- else }}
+  {{- end }}
+  {{- if or $config.http3 (eq (toString (default "TCP" $config.protocol)) "TCP") }}
```

Comment:
Why do we make the extra check here? All http3 endpoints should also contain a tcp port for the backwards http2 compatibility/upgrade-path

Reply:
This is what the template does. It adds the port to `udpPorts` if `protocol == 'UDP'`, it adds it to `tcpPorts` if `protocol == 'TCP'` and adds it to both `udpPorts` and `tcpPorts` if `http3 == true`